### PR TITLE
Add java default hash implementation of a ServerSelector for memcache

### PR DESCRIPTION
**Pull request by mdubbyap**

### CODE REVIEW

**Comment by mpetazzoni** [comment]
We really shouldn't be depending on some random GitHub repository's master branch in our build. Can we not pin the dependencies to known commit SHAs?

**Comment by mdubbyap** [comment]
Which random master are you referring to @mpetazzoni ?

File: gojavahash/gojavahash_test.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+)
+
+func TestGoJavaHash(t *testing.T) {
+	Convey("With a new JavaServerSelector", t, func() {
+		js := &JavaServerSelector{}
+		a, err := js.PickServer("test")
+		So(a, ShouldBeNil)
+		So(err, ShouldNotBeNil)
+		Convey("Add Single Server should work", func() {
+			err := js.AddServer("processdata-mc-lab2--aaaa.int.signalfuse.com:11211")
```
Comment (dgriff-tools) [inline gojavahash/gojavahash_test.go]: shouldn't the hostnames be generic names (maybe based on testcase) since this is a public repo?

**Comment by mpetazzoni** [comment]
@mdubbyap I was. Thanks. No worries then.

File: gojavahash/gojavahash_test.go (hunk 1 above)
Comment (mdubbyap) [inline gojavahash/gojavahash_test.go]: fair point, updated.

**Comment by dgriff-tools** [comment]
lgtm

**Pull request merged**